engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerService.java

```diff
@@ -0,0 +1,268 @@
+        if (images == null) {
+            return Collections.emptyList();
+        }
+        return images.stream().map(Image::getRepoTags).filter(Objects::nonNull).flatMap(Arrays::stream).collect(Collectors.toList());
+    }
+
+    public void pullImage(DockerClient client, String imageName) {
+        log.info("Downloading " + imageName + " ... this may take some time, but we only need to do it once");
+        PullImageResultCallback callback = new PullImageResultCallback() {
+            @Override
+            public void onError(Throwable throwable) {
+                log.error("Failed to pull image" + throwable.getMessage());
+                super.onError(throwable);
+            }
+        };
+        try {
+            client.pullImageCmd(imageName).exec(callback).awaitCompletion();
+        } catch (InterruptedException ex) {
+            log.error(ex.toString());
+            throw new RuntimeException("Failed to pull image" + ex.getMessage(), ex);
+        }
+    }
+
+    public DockerClient dockerClient() {
+        DefaultDockerClientConfig.Builder builder = DefaultDockerClientConfig.createDefaultConfigBuilder()
+                .withDockerTlsVerify(false);
+        Optional.ofNullable(dockerProperties.getRegistryUrl()).ifPresent(builder::withRegistryUrl);
+        Optional.ofNullable(dockerProperties.getRegistryPassword()).ifPresent(builder::withRegistryPassword);
+        Optional.ofNullable(dockerProperties.getRegistryUsername()).ifPresent(builder::withRegistryUsername);
+        DefaultDockerClientConfig config = builder.build();
+
+        URI host = Optional.ofNullable(System.getProperty("use.socket.path")).map(path ->
+                URI.create("unix://" + path)
+        ).orElseGet(config::getDockerHost);
+        DockerHttpClient dockerHttpClient = new ApacheDockerHttpClient.Builder()
+                .dockerHost(host)
+                .sslConfig(config.getSSLConfig())
+                .maxConnections(50)
+                .build();
+
+        return DockerClientImpl.getInstance(config, dockerHttpClient);
+    }
+
+    @Getter
+    public static class DockerOverseer extends AbstractOverseer {
+        private final StringBuffer stdout;
+
+        public DockerOverseer(long id, Instant started, StringBuffer stdout, CompletableFuture<ExecutionOutcome> result) {
+            super(id, (stage, out) -> {}, started, null, 0, result);
+            this.stdout = stdout;
+        }
+
+        @Override
+        public String getStdout() {
+            return stdout.toString();
+        }
+
+        @Override
+        public CompletableFuture<ExecutionOutcome> abort() {
+            return result.isDone() ? result : CompletableFuture.completedFuture(
+                    new ExecutionOutcome(Stage.ABORT, "Abort is not supported for SQL analysis", null)
+            );
+        }
+
+        @Override
+        public AnalysisRequestTypeDTO getType() {
+            return SQL;
```

Comment:
Defect: definitely not `SQL` here. Discuss, whether it should be `R` or new type. @alex-odysseus do you have an opinion on this?

Reply:
Changing it to `R`, while confirming. A new Type would be good.